File: packages/rat/crates/rat-cli/tests/fixtures/sample.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Config:
    name: str
    debug: bool = False


class Processor:
    def __init__(self, config: Config):
        self.config = config

    def run(self, data: list[str]) -> list[str]:
        return [item.upper() for item in data]


def create_processor(name: str) -> Processor:
    config = Config(name=name)
    return Processor(config)


MAX_WORKERS = 4

File: packages/rat/crates/rat-cli/tests/fixtures/sample.go
package sample

import "fmt"

const MaxRetries = 3

var DefaultTimeout = 30

type Config struct {
	Name  string
	Debug bool
}

type Processor interface {
	Process(data string) (string, error)
}

func NewConfig(name string) *Config {
	return &Config{Name: name}
}

func (c *Config) String() string {
	return fmt.Sprintf("Config{name=%s}", c.Name)
}
